dockertest/docker_daemon.py

```diff
@@ -159,16 +159,32 @@ def pid():
     if not stdout.startswith('MainPID='):
         raise RuntimeError("Unexpected output from %s: expected MainPID=NNN,"
                            " got '%s'" % (cmd_result.command, stdout))
-    return int(stdout[8:])
-
-
-def cmdline():
+    mainpid = int(stdout[8:])
+    cmd = cmdline(mainpid)
+    if 'dockerd' in cmd[0]:
+        return mainpid
+    # As of April 2017 we may be running dockerd under runc. If so, actual
+    # dockerd process is an immediate child of the one returned by systemd.
+    for child_pid in utils.run("pgrep -P %s" % mainpid).stdout.split():
```

Comment:
How about: ``for child_pid in utils.get_children(mainpid)``?  (I dunno if ``pgrep`` exists on atomic, but ``ps`` certainly does).

Also, maybe it's overkill, but maybe consider ``if not utils.pid_is_alive(mainpid): raise something()``?